A92-xds-ext-authz.md

```diff
@@ -0,0 +1,220 @@
+  could consider adding it in the future if there is a need.
+- filter_enabled, filter_enabled_metadata, deny_at_disable: We don't currently
+  support the runtime system.
+- include_tls_session: We do not currently have a use-case for this, but
+  we could consider adding it in the future if we do encounter such a
+  use-case.
+
+We will not support per-route config overrides for this filter.  TODO:
+If we need the `disabled` flag, maybe just support that via the
+`FilterConfig.disabled` field in `typed_per_filter_config` instead of
+doing it in an ext_authz-specific way?  Just make sure that the
+semantics are the same -- ext_authz config says "If disabled is
+specified in multiple per-filter-configs, the most specific one will be
+used."
+
+### Communication With the ext_authz Server
+
+The [`AttributeContext`
+message](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L43)
+sent to the server will be populated as follows:
+- [source](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L192): Will always be set.  Inside it:
+  - [address](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L58)
+    and [service](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L65)
+    will be set.
+  - labels will not be set.
+  - [principal](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L82)
+    will be set if the client provided a cert and we validates it, unset
+    otherwise.
+  - [certificate](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L86): This will be populated if configured.
+- [destination](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L197):
+  This field will not be populated, because there is no destination on a gRPC
+  server.
+- [request](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L200):
+  Will always be set.  Inside it:
+  - [time](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L95):
+    Will be set to the RPC's start time.
+  - [http](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L98):
+    Will always be set.  Inside of it:
+    - [id](https://github.com/envoyproxy/envoy/blob/cdd19052348f7f6d85910605d957ba4fe0538aec/api/envoy/service/auth/v3/attribute_context.proto#L112): TODO: how do we set this?
```

Comment:
RLQS is using the `x-request-id` request metadata (which is most likely not set).